aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/AWSAppSyncClient.java

```diff
@@ -703,10 +798,74 @@ public boolean isMutationQueueEmpty() {
     }
 
     /**
-     * Clear the mutation queue. A Mutation that is currently in progress will continue to execute until finished.
+     * Clear the mutation queue. A Mutation that is currently in progress will
+     * continue to execute until finished.
      *
+     * @deprecated Please use
+     *     #clearCache(ClearCacheOptions.builder().clearMutations().build()) instead.
      */
+    @Deprecated
     public void clearMutationQueue() {
         mAppSyncOfflineMutationManager.clearMutationQueue();
     }
+
+    /**
+     * Clear the store created for Delta Sync. This method deletes all the records
+     * that stores the lastSyncTime of the sync queries.
+     */
+    private void clearDeltaSyncStore() {
```

Comment:
What happens to ongoing delta syncs if this is cleared?

Reply:
The developer is in charge of checking the `AWSAppSyncDeltaSyncWatcher` and canceling them before calling the `clearCache()`

Reply:
Couple of suggestions to consider
1) Add the expectation to the comments
2) Check if there is an ongoing delta sync and exception out if there is an active delta sync.